src/_imaging.c

```diff
@@ -1575,11 +1575,7 @@ if (PySequence_Check(op)) { \
             }
             double value;
             if (image->bands == 1) {
-                int bigendian;
-                if (image->type == IMAGING_TYPE_SPECIAL) {
-                    // I;16*
-                    bigendian = strcmp(image->mode, "I;16B") == 0;
-                }
+                int bigendian = image->type == IMAGING_TYPE_SPECIAL && strcmp(image->mode, "I;16B") == 0;
```

Comment:
I'm for just default value.

```suggestion
                int bigendian = 0;
                if (image->type == IMAGING_TYPE_SPECIAL) {
                    bigendian = strcmp(image->mode, "I;16B") == 0;
                }
```